src/redis_rdb.cc

```diff
@@ -0,0 +1,429 @@
+}
+
+// Use the logic of redis directly
+int RedisDatabase::Util::encodeInteger(int64_t value, unsigned char *enc) {
+    if (value >= -(1 << 7) && value <= (1 << 7) - 1) {
+        enc[0] = (RDB_ENCVAL << 6) | RDB_ENC_INT8;
+        enc[1] = value & 0xFF;
+        return 2;
+    } else if (value >= -(1 << 15) && value <= (1 << 15) - 1) {
+        enc[0] = (RDB_ENCVAL << 6) | RDB_ENC_INT16;
+        enc[1] = value & 0xFF;
+        enc[2] = (value >> 8) & 0xFF;
+        return 3;
+    } else if (value >= -((int64_t)1<<31) && value <= ((int64_t)1<<31)-1) {
+        enc[0] = (RDB_ENCVAL << 6) | RDB_ENC_INT32;
+        enc[1] = value & 0xFF;
+        enc[2] = (value >> 8) & 0xFF;
+        enc[3] = (value >> 16) & 0xFF;
+        enc[4] = (value >> 24) & 0xFF;
+        return 5;
+    } else {
+        return 0;
+    }
+}
+
+// Use the logic of redis directly
+int RedisDatabase::Util::ll2string(char *dst, size_t dstlen, int64_t svalue) {
+  static const char digits[201] =
+      "0001020304050607080910111213141516171819"
+      "2021222324252627282930313233343536373839"
+      "4041424344454647484950515253545556575859"
+      "6061626364656667686970717273747576777879"
+      "8081828384858687888990919293949596979899";
+  int negative;
+  uint64_t value;
+
+  /* The main loop works with 64bit unsigned integers for simplicity, so
+     * we convert the number here and remember if it is negative. */
+  if (svalue < 0) {
+    if (svalue != LLONG_MIN) {
+      value = -svalue;
+    } else {
+      value = ((uint64_t)LLONG_MAX) + 1;
+    }
+    negative = 1;
+  } else {
+    value = svalue;
+    negative = 0;
+  }
+
+  /* Check length. */
+  uint32_t const length = digits10(value) + negative;
+  if (length >= dstlen) return 0;
+
+  /* Null term. */
+  uint32_t next = length;
+  dst[next] = '\0';
+  next--;
+  while (value >= 100) {
+    int const i = (value % 100) * 2;
+    value /= 100;
+    dst[next] = digits[i + 1];
+    dst[next - 1] = digits[i];
+    next -= 2;
+  }
+
+  /* Handle last 1-2 digits. */
+  if (value < 10) {
+    dst[next] = '0' + (uint32_t)value;
+  } else {
+    int i = (uint32_t)value * 2;
+    dst[next] = digits[i + 1];
+    dst[next - 1] = digits[i];
+  }
+
+  /* Add sign. */
+  if (negative) dst[0] = '-';
+  return length;
+}
+
+Status RedisDatabase::Util::saveObjectType(const RedisType &type) {
+    switch (type) {
+        case kRedisString:
+            return saveType(RDB_TYPE_STRING);
+        case kRedisHash:
+            return saveType(RDB_TYPE_HASH);
+        case kRedisList:
+            return saveType(RDB_TYPE_LIST);
+        case kRedisSet:
+            return saveType(RDB_TYPE_SET);
+        case kRedisZSet:
+            return saveType(RDB_TYPE_ZSET);
+        case kRedisBitmap:
+            return saveType(RDB_TYPE_STRING);
+        default:
+            return Status(Status::NotOK, "illegal type");
+    }
+}
+
+Status RedisDatabase::Util::saveSecondTime(const int32_t &remain) {
```

Comment:
```suggestion
Status RedisDatabase::Util::saveSecondTime(int32_t remain) {
```

I think this const ref is useless.